src/core/cgroup.c

```diff
@@ -987,11 +1007,17 @@ static void cgroup_apply_legacy_cpu_quota(Unit *u, usec_t quota, usec_t period)
 }
 
 static uint64_t cgroup_cpu_shares_to_weight(uint64_t shares) {
+        if (shares == CGROUP_CPU_SHARES_IDLE)
+                return CGROUP_WEIGHT_IDLE;
+
         return CLAMP(shares * CGROUP_WEIGHT_DEFAULT / CGROUP_CPU_SHARES_DEFAULT,
                      CGROUP_WEIGHT_MIN, CGROUP_WEIGHT_MAX);
 }
 
 static uint64_t cgroup_cpu_weight_to_shares(uint64_t weight) {
+        if (weight == CGROUP_WEIGHT_IDLE)
+                return CGROUP_CPU_SHARES_IDLE;
```

Comment:
does the idle stuff even exist on cgroupv1? If not, such a conversion doesn't make any sense.

Reply:
> does the idle stuff even exist on cgroupv1? If not, such a conversion doesn't make any sense.

yes, idle exists in cgroupv1

Reply:
ok, but still: we do not add new features to our cgroupsv1 support. So even if it supports it, our code should not.